Headers/state_chm_mod.F90

```diff
@@ -2482,8 +2501,8 @@ SUBROUTINE Init_Mapping_Vectors( Input_Opt, State_Chm, RC )
        State_Chm%Map_WetDep = 0
     ENDIF
 
-    IF ( W_ > 0 ) THEN
-       ALLOCATE( State_Chm%Map_WL( W_ ), STAT=RC )
+    IF ( State_Chm%Phot%nWLbins > 0 ) THEN
+       ALLOCATE( State_Chm%Map_WL( State_Chm%Phot%nWLbins ), STAT=RC )
```

Comment:
This line (and similar lines below) makes use of a state variable field instead of a hardwired parameter.